webpack.config.mjs

```diff
@@ -49,20 +59,25 @@ export default (env) => {
         runLint: false,
         sourceDir: path.resolve(__dirname, "build"),
         startUrl: "http://localhost:3000",
-        target: env.TARGET,
+        target: WEB_EXT_TARGETS[target],
       })
     );
   }
 
+  const entry =
```

Comment:
Curious whether we've also tried shipping the same service_worker script at `background.scripts` for FF and using v3 for both?

Reply:
I definitely went back and forth on this and I'm still a bit ensure whether I've made the right choice to split them. I was mostly thinking about this from the perspective that background scripts and service workers have [some differences](https://developer.chrome.com/docs/extensions/develop/migrate/to-service-workers) and if we needed to do something in one that isn't supported in the other, I didn't want to break one of the browsers. It's a future-proofing essentially.

That said, we don't have these differences now which means both the background script and service worker code is duplicated. This increases the maintenance burden when we _do_ need to make updates. In fact, I've started working on [updating](https://github.com/apollographql/apollo-client-devtools/pull/1226) the message passing between all the various pieces of the extension, and this problem is very apparent since I need to copy/paste all changes from one to the other.

My initial thinking with just moving Chrome to v3 and leaving firefox as-is was to ensure I didn't introduce a potential breakage in both browsers simultaneously. But, this PR ended up being less change than I initially thought, so perhaps we should just bite the bullet and update both.

(sorry for the rambling... just wanted to share my thought process here 😆)

Reply:
Pushed an update that makes the code sharing a bit more robust: [663c402](https://github.com/apollographql/apollo-client-devtools/pull/1225/commits/663c4025a64638e662b842529086e087b04b935e). No need to copy/paste code, just import and use the existing background script!

I like this as a happy medium because I think the separate files highlights the differences in the two manifest/browser versions.

Thoughts on this?

Reply:
Firefox on manifest v3 is proving to be a bit trickier. The message passing seems to break. I think I'd like to look at this in a separate PR to figure out what's going on.